Src/EventCounterCollector/EventCounterCollector/EventCounterListener.cs

```diff
@@ -0,0 +1,230 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.EventCounterCollector.Implementation
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Diagnostics.Tracing;
+    using System.Globalization;
+    using System.Text;
+    using Microsoft.ApplicationInsights.DataContracts;
+
+    /// <summary>
+    /// Implementation to listen to EventCounters.
+    /// </summary>
+    internal class EventCounterListener : EventListener
+    {        
+        private readonly string refreshIntervalInSecs;
+        private readonly double refreshInternalInSecDouble;
+        private readonly EventLevel level = EventLevel.Critical;
+        private bool isInitialized = false;
+        private TelemetryClient telemetryClient;
+        private Dictionary<string, string> refreshIntervalDictionary;
+
+        // Thread-safe variable to hold the list of all EventSourcesCreated.
+        // This class may not be instantiated at the time of EventSource creation, so the list of EventSources should be stored to be enabled after initialization.
+        private ConcurrentQueue<EventSource> allEventSourcesCreated;        
+
+        // EventSourceNames from which counters are to be collected are the keys for this IDictionary.
+        // The value will be the corresponding ICollection of counter names.
+        private IDictionary<string, ICollection<string>> countersToCollect = new Dictionary<string, ICollection<string>>();
+
+        public EventCounterListener(TelemetryClient telemetryClient, IList<EventCounterCollectionRequest> eventCounterCollectionRequests, double refreshIntervalSecs)
```

Comment:
`refreshIntervalSec` shouldn't be a double. Internally we do parse it as float - https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/CounterGroup.cs#L57 - but I'd suggest using integer for this since we are planning on removing that undocumented behavior (of parsing it as a float).